Add support for Campaigns for Apps

tzengerink opened the pull request:
Thanks to @gvisniuc the Python client is close to supporting Campaigns for Apps 🎉 

@Rifinio, @spirosikmd and @shuisman could you guys review the changes?

Commit:
Added campaigns for apps support and examples

Commit:
Add Campaigns for Apps to the Readme

The Readme did not reference the Campaigns for Apps functionality yet.
Also the headers within the Readme were not rendered properly.

spirosikmd reviewed (approved):
LGTM! Would be nice to set up tests!

Rifinio reviewed (approved):
Nice

Commit:
Add tests for the added resources

Added a couple of tests to make sure the newly added resources are
considered valid. While doing so did a minor refactor.

Commit:
Bump up the version

After adding support for Campaigns for Apps a new version has to be
tagged and published.
